src/Types.hs

```diff
@@ -452,33 +452,36 @@ veName f (Tile a b) = fmap (\b' -> Tile a b') (f b)
 veState :: Lens' Tile ViewState
 veState f (Tile a b) = fmap (\a' -> Tile a' b) (f a)
 
+type Layers = V.Vector Layer
+
 data View = View
     { _vFocus :: Name
-    , _vWidgets :: Tiles
+    , _vLayers :: Layers
     }
 
--- A tile is a view element with a widget and it's view state
-newtype Tiles =
-  Tiles (V.Vector Tile)
-  deriving (Eq, Show)
+vLayers :: Lens' View Layers
+vLayers = lens _vLayers (\settings x -> settings { _vLayers = x })
 
-type instance Index Tiles = Name
-type instance IxValue Tiles = Tile
+vFocus :: Lens' View Name
+vFocus = lens _vFocus (\settings x -> settings { _vFocus = x})
 
-instance Ixed Tiles where
-  ix = tile
+-- A layer is a view element with a list of widgets and their view
+-- state
+newtype Layer =
+  Layer (V.Vector Tile)
+  deriving (Eq, Show)
 
-tileiso :: Iso' Tiles (V.Vector Tile)
-tileiso = iso (\(Tiles xs) -> xs) Tiles
+type instance Index Layer = Name
+type instance IxValue Layer = Tile
 
-tile :: Name -> Traversal' Tiles Tile
-tile k = tileiso . traversed . filtered (\x -> k == view veName x)
+instance Ixed Layer where
+  ix = tile
 
-vWidgets :: Lens' View Tiles
-vWidgets = lens _vWidgets (\settings x -> settings { _vWidgets = x })
+layeriso :: Iso' Layer (V.Vector Tile)
+layeriso = iso (\(Layer xs) -> xs) Layer
```

Comment:
Don't bother changing the patch, but I just want to draw your attention to http://hackage.haskell.org/package/lens-4.17.1/docs/Control-Lens-Iso.html#v:coerced which is an automatic iso between newtypes and the underlying type.

Reply:
I have trouble with the typing tho. For example:

```Haskell
tile :: Name -> Traversal' Layer Tile
tile k = layeriso . traversed . filtered (\x -> k == view veName x)
```

If I replace `layeriso` with `coerced` I get:

```
src/Types.hs:484:10-16: error:
    • Couldn't match representation of type ‘V.Vector Tile’
                               with that of ‘f0 Tile’
        arising from a use of ‘coerced’
    • In the first argument of ‘(.)’, namely ‘coerced’
      In the expression:
        coerced . traversed . filtered (\ x -> k == view veName x)
      In an equation for ‘tile’:
          tile k = coerced . traversed . filtered (\ x -> k == view veName x)
    |
484 | tile k = coerced . traversed . filtered (\x -> k == view veName x)
    |          ^^^^^^^
```

If I use coerced for the `layeriso` implementation everything checks out. What's the better way to apply it in this situation?